src/drawer.js

```diff
@@ -86,39 +87,46 @@ export default class Drawer extends util.Observer {
     handleEvent(e, noPrevent) {
         !noPrevent && e.preventDefault();
 
-        const clientX = e.targetTouches
-            ? e.targetTouches[0].clientX
-            : e.clientX;
+        const clientX = util.withOrientation(
+            e.targetTouches ? e.targetTouches[0] : e,
+            this.params.vertical
+        ).clientX;
         const bbox = this.wrapper.getBoundingClientRect();
 
         const nominalWidth = this.width;
         const parentWidth = this.getWidth();
+        const progressPixels = this.getProgressPixels(bbox, clientX);
 
         let progress;
         if (!this.params.fillParent && nominalWidth < parentWidth) {
-            progress =
-                (this.params.rtl ? bbox.right - clientX : clientX - bbox.left) *
-                    (this.params.pixelRatio / nominalWidth) || 0;
+            progress = progressPixels *
+                (this.params.pixelRatio / nominalWidth) || 0;
         } else {
-            progress =
-                ((this.params.rtl
-                    ? bbox.right - clientX
-                    : clientX - bbox.left) +
-                    this.wrapper.scrollLeft) /
-                    this.wrapper.scrollWidth || 0;
+            progress = (progressPixels + this.wrapper.scrollLeft) /
+                this.wrapper.scrollWidth || 0;
         }
 
         return util.clamp(progress, 0, 1);
     }
 
+    getProgressPixels(wrapperBbox, clientX) {
+        if (this.params.rtl) {
+            return wrapperBbox.right - clientX;
+        } else {
+            return clientX - wrapperBbox.left;
+        }
+    }
+
     setupWrapperEvents() {
         this.wrapper.addEventListener('click', e => {
```

Comment:
`e` might as well be called `event`

Reply:
`e` seems to be used everywhere else in the project. Happy to change this one but it will be inconsistent with what seems to be the convention.